solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -318,6 +323,37 @@ contract InterchainAccountRouter is Router {
         }
     }
 
+    function moduleType() external pure returns (uint8) {
+        return uint8(IInterchainSecurityModule.Types.ROUTING);
+    }
+
+    CommitmentReadIsm public immutable CCIP_READ_ISM;
+
+    function verify(
+        bytes calldata _metadata,
+        bytes calldata _message
+    ) external returns (bool) {
+        bytes calldata _body = _message.body();
+        IInterchainSecurityModule _ism = IInterchainSecurityModule(
+            _body.ism().bytes32ToAddress()
+        );
+
+        // If the ISM is not set, we need to check if the message is a reveal
+        // If it is, we need to set the ISM to the CCIP read ISM
+        // Otherwise, we need to set the ISM to the default ISM
+        if (address(_ism) == address(0)) {
+            if (
+                _body.messageType() ==
+                InterchainAccountMessage.MessageType.REVEAL
+            ) {
+                _ism = CCIP_READ_ISM;
+            } else {
+                _ism = mailbox.defaultIsm();
+            }
+        }
+        return _ism.verify(_metadata, _message);
```

Comment:
this logic replaces `InterchainAccountISM` yes?
we should remove that contract

Reply:
Done